### PR TITLE
 buildscripts,travis: fetch from mvn with retries

### PR DESCRIPTION
**Pull request by zpencer**
A band aid for #3284

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,6 +25,8 @@ before_install:
   - echo "errorProne=true" >> $HOME/.gradle/gradle.properties
 
 install:
+  # Warm mvn cache with retries, in case connection to maven repo is flakey
```
Comment (ejona86) [inline .travis.yml]: Travis has a cache, so this hasn't been a problem.

File: buildscripts/kokoro/unix.sh [hunk 2]
```diff
@@ -50,6 +50,9 @@ export LD_LIBRARY_PATH=/tmp/protobuf/lib
 export LDFLAGS=-L/tmp/protobuf/lib
 export CXXFLAGS="-I/tmp/protobuf/include"
 
+# Warm mvn cache with retries, in case connection to maven repo is flakey
+(./gradlew assemble || ./gradlew assemble || ./gradlew assemble) && ./gradlew clean
```
Comment (ejona86) [inline buildscripts/kokoro/unix.sh]: How about `./gradlew -m build` (`-m` is "dry run")? That skips the actual tasks but appears like it downloads necessary deps.

File: buildscripts/kokoro/cronet.sh [hunk 3]
```diff
@@ -5,4 +5,8 @@ cat /VERSION
 
 cd ./github/grpc-java/cronet
 ./cronet_deps.sh
+
+# Warm mvn cache with retries, in case connection to maven repo is flakey
+(../gradlew assemble || ../gradlew assemble || ../gradlew assemble) && ../gradlew clean
```
Comment (ejona86) [inline buildscripts/kokoro/cronet.sh]: These need `--include-build ..`

**Comment by zpencer** [comment]
`./gradlew -m build` does not download everything all at once:
https://sponge.corp.google.com/target?id=48069fb3-a0c7-4827-a3b7-e230fc1e8e38&target=grpc%2Fjava%2Fpresubmit%2Fmacos&searchFor=

File: build.gradle [hunk 4]
```diff
@@ -59,11 +58,12 @@ subprojects {
...
 
     repositories {
-        mavenCentral()
-        mavenLocal()
         maven {
-          url "https://oss.sonatype.org/content/repositories/snapshots/"
+            // The google mirror is less flaky than mavenCentral()
+            url "https://maven-central.storage-download.googleapis.com/repos/central/data/"
         }
+        mavenCentral()
```
Comment (ejona86) [inline build.gradle]: Why do we include `mavenCentral()` here but not in the `buildscript{}` sections? I'd expect us to be consistent, one way or the other.

File: buildscripts/kokoro/cronet.sh [hunk 5]
```diff
@@ -5,4 +5,5 @@ cat /VERSION
 
 cd ./github/grpc-java/cronet
 ./cronet_deps.sh
+
```
Comment (ejona86) [inline buildscripts/kokoro/cronet.sh]: revert?

File: gae-interop-testing/gae-jdk7/build.gradle [hunk 6]
```diff
@@ -30,7 +33,10 @@ repositories {   // repositories for Jar's you access in your code
     url 'https://maven-central.storage.googleapis.com'             // Google's mirror of Maven Central
```
Comment (ejona86) [inline gae-interop-testing/gae-jdk7/build.gradle]: This seems weird.

**Comment by zpencer** [comment]
Note for posterity: Just realized I committed this change using the original title of the PR, before deciding to change the strategy.

File: all/build.gradle [hunk 7]
```diff
@@ -4,7 +4,10 @@ description = "gRPC: All"
 
 buildscript {
     repositories {
-        mavenCentral()
+        maven {
+            // The google mirror is less flaky than mavenCentral()
+            url "https://maven-central.storage-download.googleapis.com/repos/central/data/"
+        }
```
Comment (dapengzhang0) [inline all/build.gradle]: @zpencer can this google mirror apply to android build.gradle files?

Comment (zpencer) [reply all/build.gradle]: We can try it out and see if it works. I'm not sure if there's anything special about android related packages.

**Pull request merged**